test/libsolidity/SyntaxTest.cpp

```diff
@@ -103,20 +115,30 @@ void SyntaxTest::parseAndAnalyze()
 
 void SyntaxTest::filterObtainedErrors()
 {
-	string const preamble = "pragma solidity >=0.0;\n// SPDX-License-Identifier: GPL-3.0\n";
 	for (auto const& currentError: filterErrors(compiler().errors(), true))
 	{
 		int locationStart = -1, locationEnd = -1;
 		string sourceName;
 		if (auto location = boost::get_error_info<errinfo_sourceLocation>(*currentError))
 		{
+			assert(location->source);
```

Comment:
This is removed if `NDEBUG` is defined. Do we really want to use c assert?

Reply:
I think the C assert a good thing because it lets me not worry if the check is too heavy or if I'm doing too many of them. I'd say that if we're worrying that it might actually happen, then it's not really an assertion and should be properly validated instead.

In Solidity I'm usually using `solAssert()` though because that's the prevailing style and I want to respect that, but I checked other files in this directory to see which variant to use (`solAssert()`, `assertThrow()`, etc.) and the few uses I saw were actually `assert()` so I went with it.

I can replace it with `solAssert()` if you prefer.

Reply:
I went ahead and changed them to `solAssert()` now.